hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -180,7 +180,7 @@ public KafkaOffsetGen(TypedProperties props) {
               .map(x -> new TopicPartition(x.topic(), x.partition())).collect(Collectors.toSet());
 
       // Determine the offset ranges to read from
-      if (lastCheckpointStr.isPresent()) {
+      if (lastCheckpointStr.isPresent() && !lastCheckpointStr.get().isEmpty()) {
```

Comment:
I think this may potentially hide some concerning errors.  
e.g. The delta streamer is consuming Kafka source, but a hidden bug happens and stored an empty checkpoint.  The next run will just ignore the empty checkpoint and reset to the `LATEST`. Then there will be data loss

Reply:
hi, if a hidden bug happens, the `*.commit` file will not be created, the next run will still read from the last successful commit, so no data will be lossed.

Reply:
Right. As you mentioned it is still possible that some wrong user behaviors might lead to an empty checkpoint. From a user perspective, I'd say if there is an empty checkpoint in the last commit, I will prefer to let the job fail other than automatically reset the checkpoint. Throw an exception if the checkpoint is empty would make more sense to me and let the user decide whether they wanna reset or not. WDYT?

Reply:
@lamber-ken : Also, Instead of handling the empty checkpoints only for kafka, can we handle it generically in DeltaSync (https://github.com/apache/incubator-hudi/blob/77d5b92d88d6583bdfc09e4c10ecfe7ddbb04806/hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java#L262) so that we can have an uniform handling of checkpoints across sources ?

Reply:
hi @bvaradar, add `!commitMetadata.getMetadata(CHECKPOINT_KEY).isEmpty()`.
but if we do that, the application will always throw `HoodieDeltaStreamerException`
![image](https://user-images.githubusercontent.com/20113411/76372301-ee63d700-6377-11ea-863a-21a99028dc5d.png)

Reply:
I agree with @bvaradar here. 
I tried 
```java
else if (commitMetadata.getMetadata(CHECKPOINT_KEY) != null && !commitMetadata.getMetadata(CHECKPOINT_KEY).isEmpty()) {
          resumeCheckpointStr = Option.of(commitMetadata.getMetadata(CHECKPOINT_KEY));
        }
```
The unit tests were passed.
If you mean when the checkpoint was empty, the application was throwing an Exception. I think it should the desired behavior, WDYT?

Reply:
hi @garyli1019, let's imagine a scenario that a topic with no data, the first commit will save empty checkpoint, then the second commit will always throw exception(even if we send msg to kafka). In that case, `EARLIEST` or `LATEST` will no longer has any effect.

Reply:
Prefer to pushdown the control bebavior to datasource(e.g kafka / pulsar)

Reply:
In this case I think the user could use `cfg.checkpoint = xxx` to reset the checkpoint. It would be concerning for me if the deltastreamer automatically reset the checkpoint for me and I didn't aware of it.

Reply:
Thanks, I need to think about it, wait a moment : )